src/main/java/com/google/api/generator/gapic/composer/MethodSampleCodeHelperComposer.java

```diff
@@ -0,0 +1,167 @@
+      VariableExpr clientVarExpr) {
+    return AssignmentExpr.builder()
+        .setVariableExpr(clientVarExpr.toBuilder().setIsDecl(true).build())
+        .setValueExpr(
+            MethodInvocationExpr.builder()
+                .setStaticReferenceType(clientVarExpr.variable().type())
+                .setReturnType(clientVarExpr.variable().type())
+                .setMethodName("create")
+                .build())
+        .build();
+  }
+
+  // Create a Map where key is method's argument name, and value is its VariableExpr.
+  private static Map<String, VariableExpr> createMethodArgumentsVariableExprs(
+      List<MethodArgument> arguments) {
+    return arguments.stream()
+        .collect(
+            Collectors.toMap(
+                methodArg -> methodArg.name(),
+                methodArg ->
+                    createVariableExpr(
+                        JavaStyle.toLowerCamelCase(methodArg.name()), methodArg.type())));
+  }
+
+  // Return a list of AssignmentExpr for method argument with its default value.
+  private static List<Expr> assignMethodArgumentsWithDefaultValues(
+      List<MethodArgument> arguments,
+      Map<String, VariableExpr> argVarExprs,
+      Map<String, ResourceName> resourceNames) {
+    return arguments.stream()
+        .map(
+            arg ->
+                createAssignmentExpr(
+                    argVarExprs.get(arg.name()),
+                    DefaultValueComposer.createDefaultValue(arg, resourceNames)))
+        .collect(Collectors.toList());
+  }
+
+  private static Expr createAssignExprForVariableWithClientMethod(
+      VariableExpr variableExpr,
+      VariableExpr clientVarExpr,
+      String methodName,
+      List<Expr> argumentsVarExprs) {
+    MethodInvocationExpr clientMethodInvocationExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(clientVarExpr)
+            .setMethodName(JavaStyle.toLowerCamelCase(methodName))
+            .setArguments(argumentsVarExprs)
+            .setReturnType(variableExpr.variable().type())
+            .build();
+    return AssignmentExpr.builder()
+        .setVariableExpr(variableExpr.toBuilder().setIsDecl(true).build())
+        .setValueExpr(clientMethodInvocationExpr)
+        .build();
+  }
+
+  private static String getClientName(TypeNode clientType) {
+    return JavaStyle.toLowerCamelCase(clientType.reference().name());
+  }
+
+  private static boolean isProtoEmptyType(TypeNode type) {
+    return type.reference().pakkage().equals("com.google.protobuf")
+        && type.reference().name().equals("Empty");
+  }
+
+  private static AssignmentExpr createAssignmentExpr(VariableExpr variableExpr, Expr valueExpr) {
+    return AssignmentExpr.builder()
+        .setVariableExpr(variableExpr.toBuilder().setIsDecl(true).build())
+        .setValueExpr(valueExpr)
+        .build();
+  }
+
+  private static VariableExpr createVariableExpr(String variableName, TypeNode type) {
```

Comment:
I'd prefer not to have this in the helper, even if it's used in other methods in future PRs, because (a) the logic is straightforward, and (b) it helps future readers see immediately what/how the VariableExpr is being created. The other, more complex settings for VariableExpr are also the reason we don't have a simpler wrapper in that class.

Reply:
Okay.

Reply:
Let's stick with using the full builder expression, because the reader would need to look elsewhere to see how `createVariableExpr` is setting up the variable. It's not immediately clear whether it has a decl, or a reference expression, etc. 

See also information hiding, or Law of Demeter: [article 1 (same principle applied to methods)](https://understandlegacycode.com/blog/are-wrapper-functions-bad-design/), [article 2](https://dzone.com/articles/the-genius-of-the-law-of-demeter)

Reply:
Thanks for sharing your thought. SG